nexus/lib/nexus/peer.ex

```diff
@@ -106,10 +115,17 @@ defmodule Nexus.Peer do
     Process.monitor(pc)
     Logger.debug("Starting peer connection #{inspect(pc)}")
 
-    state = %State{
+    Process.link(channel)
+
+    state = %{
       id: id,
       channel: channel,
-      pc: pc
+      pc: pc,
+      inbound_tracks: %{video: nil, audio: nil},
+      outbound_tracks: %{},
+      peer_tracks: %{},
+      pending_notifications: [],
+      scheduled_subscriptions: []
```

Comment:
I think it would be easier to replace those two fields with a MapSet of `pending_peers`. As long as there is negotiation in-progress, just modify the set by adding or removing peers. When the negotiation finishes, add pendning peers. This way we can get rid of generic `pending_notificaitons` and very detailed `scheduled_subscriptions`